sootup.callgraph/src/main/java/sootup/callgraph/GraphBasedCallGraph.java

```diff
@@ -129,6 +130,84 @@ public int callCount() {
     return graph.edgeSet().size();
   }
 
+  @Override
+  public void convertToDotFormatAndFileWrite(String filePath) {
+    StringBuilder dotFormatBuilder = new StringBuilder();
+
+    for (Edge edge : graph.edgeSet()) {
+      Vertex sourceVertex = graph.getEdgeSource(edge);
+      Vertex targetVertex = graph.getEdgeTarget(edge);
+      dotFormatBuilder
+          .append("\t")
+          .append(sourceVertex.methodSignature)
+          .append(" -> ")
+          .append(targetVertex.methodSignature)
+          .append(";\n");
+    }
+    // Sort the callGraph
+    String sortedCallGraph = sortCallGraph(dotFormatBuilder.toString());
```

Comment:
dont sort the string. Sort the edges in the call graph it is way easier

Reply:
wasting execution time to extract strings if we have already ways to get them before we call tostring on the method signature